assets/src/edit-story/components/canvas/canvasLayout.js

```diff
@@ -1,62 +1,223 @@
  */
-import { CENTRAL_RIGHT_PADDING, LEFT_NAV_WIDTH, PAGE_WIDTH, PAGE_HEIGHT, PAGE_NAV_PADDING } from '../../constants';
-import Page from './page';
+import { LEFT_NAV_WIDTH, PAGE_WIDTH, PAGE_HEIGHT, PAGE_NAV_BUTTON_WIDTH } from '../../constants';
+import { useStory } from '../../app';
+import useCanvas from './useCanvas';
+import PageDisplay from './page';
+import PageFrames from './pageFrames';
+import PageEdit from './pageEdit';
 import PageMenu from './pagemenu';
 import PageNav from './pagenav';
 import Carrousel from './carrousel';
 import SelectionCanvas from './selectionCanvas';
 
-const Background = styled.div`
+// QQQQ: remove.
+const DEBUG_ANGLE = 0;
+
+const WorkspaceBackground = styled.div`
 	background-color: ${ ( { theme } ) => theme.colors.bg.v1 };
-	display: flex;
-	flex-direction: column;
-	align-items: center;
-	justify-content: center;
+	width: 100%;
 	height: 100%;
+	position: relative;
+	user-select: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: darkgray;
+		backface-visibility: hidden;
+		transform-style: preserve-3d;
+		transform-origin: 50% 50%;
+		transform: perspective(2500px) scale(0.4) rotateY(${ DEBUG_ANGLE }deg);
+	` ) }
+`;
+
+const Layer = css`
+	position: absolute;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+`;
 
+// @todo: the menu and carousel heights are not correct until we make a var-size
+// page.
+const Layout = css`
 	display: grid;
 	grid:
-    ".    .      .          . .        ." 1fr
-    ".    prev   page       . next     ." ${ PAGE_HEIGHT }px
-    ".    .      menu       . .        ." 48px
-    ".    .      carrousel  . .        ." 1fr
-    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_PADDING }px 1fr ${ CENTRAL_RIGHT_PADDING }px;
+    ".    .      . .          . .        ." 1fr
+    ".    prev   . page       . next     ." ${ PAGE_HEIGHT }px
+    ".    .      . menu       . .        ." 48px
+    ".    .      . .          . .        ." 1fr
+    ".    carrousel      carrousel carrousel  carrousel carrousel        ." 60px
+    / 1fr ${ LEFT_NAV_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ PAGE_WIDTH }px ${ PAGE_NAV_BUTTON_WIDTH }px ${ LEFT_NAV_WIDTH }px 1fr;
+`;
+
+const Canvas = css`
+	position: relative;
+	width: 100%;
+	height: 100%;
+`;
+
+const DisplayLayer = styled.div`
+	${ Layer }
+	${ Layout }
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 255, 0, 0.2);
+		transform: translateZ(200px);
+	` ) }
+`;
+
+const DisplayCanvas = styled.div`
+	${ Canvas }
+	grid-area: page;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(255, 255, 255, 0.2);
+	` ) }
+`;
+
+const NavLayer = styled.div`
+	${ Layer }
+	${ Layout }
+	pointer-events: none;
+
+	${ DEBUG_ANGLE && ( `
+		background-color: rgba(0, 0, 255, 0.2);
+		transform: translateZ(400px);
+	` ) }
 `;
 
-const Area = styled.div`
+const NavArea = styled.div`
+	background-color: ${ ( { theme } ) => theme.colors.bg.v1 };
 	grid-area: ${ ( { area } ) => area };
 	height: 100%;
 	width: 100%;
+	pointer-events: initial;
```

Comment:
why is this needed? or just for readability?

Reply:
Layers are normally `pointer-events: none` - they are only used for `z-index` in terms of display, clipping, and event propagation - but otherwise do not directly respond to events. The children of layers actually handle events, so the `pointer-events` is restored.

Reply:
To add: I've been trying to keep this `pointer-events` off -> on logic in this one module. Not yet sure it'd work like this for everything.

Reply:
oh I missed that these are children. All good then.